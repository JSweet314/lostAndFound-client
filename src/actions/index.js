export const captureUser = user => ({
  type: 'CAPTURE_USER',
  user
});

export const logOutUser = () => ({
  type: 'LOG_OUT_USER'
});

export const submitNewUser = user => ({
  type: 'SUBMIT_NEW_USER',
  user
});

export const captureErrorMessage = message => ({
  type: 'CAPTURE_ERROR_MESSAGE',
  message
});

export const clearErrorMessage = () => ({
  type: 'CLEAR_ERROR_MESSAGE'
});