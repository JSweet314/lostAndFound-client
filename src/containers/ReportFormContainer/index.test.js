import React from 'react';
import { shallow } from 'enzyme';
import * as actions from '../../actions';
import { ReportFormContainer, mapDispatchToProps, mapStateToProps }
  from './index';

describe('ReportFormContainer', () => {
  let wrapper;
  const mockMatch = { params: { id: 'lost' } };
  const mockState = {
    name: 'phone',
    description: 'black iphone',
    location: 'Turing School',
    date: '04/01/2018',
    reward: '0'
  };
  const mockItem = { ...mockState, status: 'lost', userId: 1 };
  const mockGoBack = jest.fn();
  const mockHistory = { goBack: () => mockGoBack() };
  const mockReportItem = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <ReportFormContainer
        userId={1}
        match={mockMatch}
        history={mockHistory}
        reportItem={mockReportItem} />
    );
  });

  it('should match a snapshot()', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a handleOnChange method', () => {
    wrapper.instance().handleOnChange({
      target: {
        name: 'name',
        value: 'phone'
      }
    });
    expect(wrapper.state('name')).toEqual('phone');
  });

  it('should have a handleGoBack method', () => {
    wrapper.instance().handleGoBack({ preventDefault: jest.fn() });
    expect(mockGoBack).toHaveBeenCalled();
  });

  it('should have a handleOnSubmit method', () => {
    wrapper.instance().handleOnSubmit({ preventDefault: jest.fn() });
    const expected = {
      name: '',
      description: '',
      location: '',
      date: '',
      reward: '',
      status: 'lost',
      userId: 1
    };
    expect(mockReportItem).toHaveBeenCalledWith(expected);
  });

  it('should reset state after submitting an item report', () => {
    wrapper.setState(mockState);
    wrapper.instance().handleOnSubmit({ preventDefault: jest.fn() });
    expect(wrapper.state()).toEqual({
      name: '',
      description: '',
      location: '',
      date: '',
      reward: ''
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with reportItem action creator', () => {
      const mockDispatch = jest.fn();
      const mapped = mapDispatchToProps(mockDispatch);
      mapped.reportItem(mockItem);
      expect(mockDispatch).toHaveBeenCalledWith(actions.reportItem(mockItem));
    });
  });

  describe('mapStateToProps', () => {
    const expected = { userId: 1 };
    expect(mapStateToProps({ user: { id: 1 } })).toEqual(expected);
  });
});