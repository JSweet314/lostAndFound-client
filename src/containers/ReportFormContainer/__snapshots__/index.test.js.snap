// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportFormContainer should match a snapshot() 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReportFormContainer
    history={
        Object {
            "goBack": [Function],
          }
    }
    match={
        Object {
            "params": Object {
              "id": "lost",
            },
          }
    }
    reportItem={[Function]}
    userId={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ReportForm
        date=""
        description=""
        handleGoBack={[Function]}
        handleOnChange={[Function]}
        handleOnSubmit={[Function]}
        location={
                Object {
                        "name": "",
                        "position": Object {
                          "lat": null,
                          "lng": null,
                        },
                      }
        }
        name=""
        reward=""
        routeId="lost"
        status=""
/>,
      "className": "report-form-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "date": "",
        "description": "",
        "handleGoBack": [Function],
        "handleOnChange": [Function],
        "handleOnSubmit": [Function],
        "location": Object {
          "name": "",
          "position": Object {
            "lat": null,
            "lng": null,
          },
        },
        "name": "",
        "reward": "",
        "routeId": "lost",
        "status": "",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ReportForm
          date=""
          description=""
          handleGoBack={[Function]}
          handleOnChange={[Function]}
          handleOnSubmit={[Function]}
          location={
                    Object {
                              "name": "",
                              "position": Object {
                                "lat": null,
                                "lng": null,
                              },
                            }
          }
          name=""
          reward=""
          routeId="lost"
          status=""
/>,
        "className": "report-form-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "date": "",
          "description": "",
          "handleGoBack": [Function],
          "handleOnChange": [Function],
          "handleOnSubmit": [Function],
          "location": Object {
            "name": "",
            "position": Object {
              "lat": null,
              "lng": null,
            },
          },
          "name": "",
          "reward": "",
          "routeId": "lost",
          "status": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
